microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshot-schema.md

```diff
@@ -0,0 +1,314 @@
+---
+title: Understand the heap snapshot file format
+description: How the heap snapshot files are structured, when exported from the Memory tool
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.date: 9/16/2022
+---
+# Understand the heap snapshot file format
+
+Investigating memory usage in web applications can be hard. The DevTools **Memory** tool allows you to explore all the objects allocated in memory by your web application by taking a heap snapshot. This information is useful for performance investigations because you can find out what objects are consuming the most memory.
+
+However, sometimes you may need to focus on specific parts of the data that the **Memory** tool doesn't show. Fortunately DevTools makes it possible to export the entire memory data as a `heapsnapshot` file which you can then analyze and visualize with your own tools.
+
+This article describes the structure and content of `heapsnapshot` files so you can build your own analysis tools.
```

Comment:
```suggestion
This article describes the structure and content of `.heapsnapshot` files so you can build your own analysis tools.
```

Reply:
I don't think anyone will want to read these files directly. This doc is really for people to build their own tools. A heapsnapshot file is tens of thousands of lines long, with basically just numbers.